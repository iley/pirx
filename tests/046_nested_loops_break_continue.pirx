extern func main(): int {
  var i: int = 0;
  var j: int;
  
  printf("Testing nested loops with break and continue:\n");
  while i < 3 {
    printf("Outer loop i=%d\n", i);
    j = 0;
    while j < 5 {
      if j == 2 {
        printf("  Breaking inner loop at j=%d\n", j);
        break;
      }
      printf("  Inner loop j=%d\n", j);
      j++;
    }
    i++;
  }
  
  printf("\n");
  
  // Test continue in nested loop
  i = 0;
  while i < 3 {
    printf("Outer loop i=%d\n", i);
    j = 0;
    while j < 4 {
      if j == 1 {
        printf("  Continuing inner loop at j=%d\n", j);
        j++;
        continue;
      }
      printf("  Inner loop j=%d\n", j);
      j++;
    }
    i++;
  }
  
  return 0;
}