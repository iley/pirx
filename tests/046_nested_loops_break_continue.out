Testing nested loops with break and continue:
Outer loop i=0
  Inner loop j=0
  Inner loop j=1
  Breaking inner loop at j=2
Outer loop i=1
  Inner loop j=0
  Inner loop j=1
  Breaking inner loop at j=2
Outer loop i=2
  Inner loop j=0
  Inner loop j=1
  Breaking inner loop at j=2

Outer loop i=0
  Inner loop j=0
  Continuing inner loop at j=1
  Inner loop j=2
  Inner loop j=3
Outer loop i=1
  Inner loop j=0
  Continuing inner loop at j=1
  Inner loop j=2
  Inner loop j=3
Outer loop i=2
  Inner loop j=0
  Continuing inner loop at j=1
  Inner loop j=2
  Inner loop j=3
